annotated.go

```diff
@@ -63,7 +63,11 @@ type Annotated struct {
 	// If specified, this will be used as the group name for all non-error values returned
 	// by the constructor. For more information on value groups, see the package documentation.
 	//
-	// A group option may not be provided if a name option is provided.
+	// A group option may not be provided if a name option is provided. Like in
+	// group tags, the group name may be followed by a `,flatten` option to
+	// indicate that each element returned from the target constructor should be
+	// injected into the container individually rather than the whole group at
+	// once.
```

Comment:
nit: different paragraphs for different points

```suggestion
	// A group option may not be provided if a name option is provided.
	//
	// Similar to group tags, the group name may be followed by a `,flatten`
	// option to indicate that each element in the slice returned by the
	// constructor should be injected into the value group individually.
```